### ISSUE TITLE
Permission denied! Please run script with sudo privileges!

### ISSUE THREAD

**Issue by abdullahcicekli** [kdrypr/CTI-Feed-Collector#1]
I gave sudo permission but the script doesn't work. I got the same error on different operating systems. What should I do ?

### MACOS 

![image](https://user-images.githubusercontent.com/48344066/140729814-03070f0f-8193-408d-826a-81547490e05c.png)

```
Do you want to install Elasticsearch? [Y/n]: n
Do you want to install Kibana? [Y/n]: n
Permission denied! Please run script with sudo privileges!

There is no data on 2021-11-08 13:53:57.659693+03:00 in voip
Do you want to import all cti json feeds to elasticsearch? [Y/n]: Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/urllib3/connection.py", line 159, in _new_conn
    conn = connection.create_connection(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/urllib3/util/connection.py", line 84, in create_connection
    raise err
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/urllib3/util/connection.py", line 74, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/elasticsearch/connection/http_urllib3.py", line 245, in perform_request
    response = self.pool.urlopen(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 726, in urlopen
    retries = retries.increment(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/urllib3/util/retry.py", line 386, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/urllib3/packages/six.py", line 735, in reraise
    raise value
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 670, in urlopen
    httplib_response = self._make_request(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/urllib3/connectionpool.py", line 392, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/http/client.py", line 1279, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/http/client.py", line 1325, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/http/client.py", line 1274, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/http/client.py", line 1034, in _send_output
    self.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/http/client.py", line 974, in send
    self.connect()
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/urllib3/connection.py", line 187, in connect
    conn = self._new_conn()
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/urllib3/connection.py", line 171, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x1060decd0>: Failed to establish a new connection: [Errno 61] Connection refused

During handling of the above exception, another exception occurred:


Traceback (most recent call last):
  File "/Users/abdullahcicekli/CTI-Feed-Collector/cti_collector.py", line 319, in <module>
    postElastic()
  File "/Users/abdullahcicekli/CTI-Feed-Collector/cti_collector.py", line 82, in postElastic
    es.indices.create(index=elasticIndex.lower(), ignore=400)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/elasticsearch/client/utils.py", line 152, in _wrapped
    return func(*args, params=params, headers=headers, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/elasticsearch/client/indices.py", line 119, in create
    return self.transport.perform_request(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/elasticsearch/transport.py", line 390, in perform_request
    raise e
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/elasticsearch/transport.py", line 358, in perform_request
    status, headers_response, data = connection.perform_request(
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/elasticsearch/connection/http_urllib3.py", line 258, in perform_request
    raise ConnectionError("N/A", str(e), e)
elasticsearch.exceptions.ConnectionError: ConnectionError(<urllib3.connection.HTTPConnection object at 0x1060decd0>: Failed to establish a new connection: [Errno 61] Connection refused) caused by: NewConnectionError(<urllib3.connection.HTTPConnection object at 0x1060decd0>: Failed to establish a new connection: [Errno 61] Connection refused)
```

### LINUX

![image](https://user-images.githubusercontent.com/48344066/140731699-b015571f-4935-4821-a31d-275fbb51e3c0.png)

```
Permission denied! Please run script with sudo privileges!
Traceback (most recent call last):
  File "cti_collector.py", line 316, in <module>
    feedService()
  File "cti_collector.py", line 299, in feedService
    blocklistDE()
  File "cti_collector.py", line 274, in blocklistDE
    data = urlopen(req).read().decode('utf-8')
  File "/usr/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 503: Service Unavailable
```

**Comment by kdrypr** [comment]
Script was configured for ubuntu. ELK stack download link has changed. You can check and edit. You can see the error when you read it.

**Comment by abdullahcicekli** [comment]
I can't see any change in the code. Are you sure you pushed it ?

![Screen Shot 2021-11-08 at 15 09 40](https://user-images.githubusercontent.com/48344066/140739649-aff836db-303a-468b-ab85-9a9f59b8c99b.png)

**Comment by kdrypr** [comment]
I said you can change. I didnt fix my code. ELK download link changes frequently

**Comment by abdullahcicekli** [comment]
I am sorry. got it a little late. Ok. I will try.

**Comment by abdullahcicekli** [comment]
I understand the problem. I got stuck with the try-except block in configureServices() function because I selected "no" in elastic and kibana setup options. For this, when we return the results of installKibana() and installElasticsearch() to the configureServices() function, the error disappears. So if the option is no, the function should not work. The same should be added to the postElastic() function. Could it be better this way? What about?

**FOR configureServices() :** 
![image](https://user-images.githubusercontent.com/48344066/140770295-00078347-1a98-4c7a-ac00-6d9404ca4c67.png)

**For postElastic() :**
![image](https://user-images.githubusercontent.com/48344066/140770633-032b0a1e-9e0d-4fae-b13e-4a7e7a9a3cf0.png)

```
p1 = click.confirm('Do you want to install Elasticsearch?', default=True)
p2 = click.confirm('Do you want to install Kibana?', default=True)
```
